chain/testing.go

```diff
@@ -31,7 +31,9 @@ type Builder struct {
 	stateBuilder StateBuilder
 	seq          uint64 // For unique tickets
 
-	blocks map[cid.Cid]*types.Block
+	blocks   map[cid.Cid]*types.Block
+	messages map[cid.Cid][]*types.SignedMessage
+	receipts map[cid.Cid][]*types.MessageReceipt
 	// Cache of the state root CID computed for each tipset key.
 	tipStateCids map[string]cid.Cid
 }
```

Comment:
Blocking (can't comment on the exact line): 
Since the `Builder` implements `MessageReader` we should:
```go
var _ MessageReader = (*Builder)(nil)
```